DD-1041: Describe v1.0.0 of the Dans Bagit Profile (#28)
DANS-KNAW/dans-bagit-profile: docs/versions/1.0.0.md

### Scope

This document specifies what constitutes an acceptable DANS bag. This includes all the requirements for a bag to be

> successfully processed by the ingest workflow. What is acceptable as a [DEPOSIT] bag is sometimes not exactly the same as what

Reviewer: ```suggestion
successfully processed by the ingest workflow. What is acceptable as a DEPOSIT bag is sometimes not exactly the same as what
```

Reviewer: [ and ] is used for links defined elsewhere.

an element in a certain namespace is intended. The table below lists the mapping from prefix to namespace. In the actual
document, the namespace may be bound to a different prefix, or be the default namespace.

| Prefix    | Namespace URI                                                    | Namespace documentation                                                 |
|-----------|------------------------------------------------------------------|-------------------------------------------------------------------------|
| `am`      | `http://easy.dans.knaw.nl/schemas/bag/metadata/agreements/`      | [DANS deposit agreement metadata]{:target=_blank}                       |
| `amd`     | `http://easy.dans.knaw.nl/easy/dataset-administrative-metadata/` | [DANS EASY administrative metadata schema]{:target=_blank}              |              
| `dc`      | `http://purl.org/dc/elements/1.1/`                               | [DC]{:target=_blank}                                                    |     
| `dcterms` | `http://purl.org/dc/terms/`                                      | [DCTERMS]{:target=_blank}                                               |    

> | `dcx-dai` | `http://easy.dans.knaw.nl/schemas/dcx/dai/`                      | [DANS DCTERMS extension: Digital Author Identifier**]**{:target=_blank} | 

Reviewer: ```suggestion
| `dcx-dai` | `http://easy.dans.knaw.nl/schemas/dcx/dai/`                      | [DANS DCTERMS extension: Digital Author Identifier]{:target=_blank} | 
```

| `dcx-gml` | `http://easy.dans.knaw.nl/schemas/dcx/gml/`                      | [DANS DCTERMS extension: Geography Markup Language]{:target=_blank}     |
| `ddm`     | `http://easy.dans.knaw.nl/schemas/md/ddm/`                       | [DANS dataset metadata schema]{:target=_blank}                          |
| `emd`     | `http://easy.dans.knaw.nl/easy/easymetadata/`                    | [DANS EASY metadata schema]{:target=_blank}                             |
| `files`   | `http://easy.dans.knaw.nl/schemas/bag/metadata/files/`           | [DANS bag file metadata schema]{:target=_blank}                         |
| `gml`     | `http://www.opengis.net/gml`                                     | [GML 3.1.1 Simplified Features profile Levels 0 and 1]{:target=_blank}  |
| `id-type` | `http://easy.dans.knaw.nl/schemas/vocab/identifier-type/`        | DANS controlled list of `dcterms:identifier` types                      |

> | `wfs`     | `http://easy.dans.knaw.nl/easy/workflow/`                        | [EASY administrative workflow metadata schema]{:target=_blank}          |

Reviewer: ```suggestion
| `wfs`     | `http://easy.dans.knaw.nl/easy/workflow/`                        | [DANS EASY administrative workflow metadata schema]{:target=_blank}          |
```

used in the [Requirements](#requirements) section of this document. Where terms are used in the definitions of other
terms, they are printed in *italics*.

* **bag**: a data package conforming to the [BagIt]{:target=_blank} specifications.

> * **bag-ID**: a [urn:uuid] given to every deposit upon delivery to DANS. The bag-ID of the first version of a dataset is also known as the *sword token*.

Reviewer: ```suggestion
* **bag-ID**: a [urn:uuid]{:target=_blank} given to every deposit upon delivery to DANS. The bag-ID of the first version of a dataset is also known as the *sword token*.
```

#### 1.1 Validity

1. The bag MUST be `VALID` according to the [BagIt]{:target=_blank} specifications v1.0 (RFC 8493).

#### 1.2 `bag-info.txt`

1. The bag MUST contain a `bag-info.txt` file.

2. (a) The `bag-info.txt` file MUST contain exactly one element called `Created`. (b) It MUST have a timestamp value
   in [ISO 8601]{:target=_blank} format, including the time zone and a millisecond precision time part. (c) The values
   in the `Created` timestamps SHOULD reflect the correct order of the versions.

3. The `bag-info.txt` file MAY contain at most one element called `Is-Version-Of` with a
   [urn:uuid]{:target=_blank}-value.

4. (a) The `bag-info.txt` file MAY contain at most one element called `Has-Organizational-Identifier`. The value of this 
   element MUST start with a client specific prefix agreed between the client and DANS. If there already exists 
   a dataset in the targeted Data Station with the same value in the field `dansOtherId` then Is-Version-Of as described 

>    in 1.2.5 <!-- TODO: check numbering --> MUST be specified to indicate that this deposit is an update of that dataset. 

Reviewer: ```suggestion
   in 1.2.3 MUST be specified to indicate that this deposit is an update of that dataset. 
```

   (b) If this `Has-Organizational-Identifier` is given, at most one `Has-Organizational-Identifier-Version` MAY be present, 
   containing the version number.

#### 1.3 Manifests

1. If the bag has only one payload manifest it MUST NOT use the MD5 algorithm. It MAY have an MD5 payload manifest in 
   addition to other payload manifests.

### 2 Structural requirements

> 1. The bag MUST have a tag-directory called `metadata` (one word, lowercase letters) directly under the bag base

Reviewer: ```suggestion
1. The bag MUST have a tag-directory called `metadata` directly under the bag base
```

9. All URLs used in `metadata/dataset.xml` MUST be valid URLs with protocol http or https.

10. The file `metadata/dataset.xml` MUST have a rights holder, either **(DEPOSIT)** in `<dcterms:rightsHolder>` or **(MIGRATION)** as an author with
    role `RightsHolder`.

#### 3.2 `metadata/files.xml`

1. The file `metadata/files.xml` MUST adhere to [DANS bag file metadata schema]{:target=_blank}.

2. Each `file` element's `filepath` attribute MUST contain the bag local path to the payload file described. When an 

>    `original-filepaths.txt` exists, rule 2.7.2<!-- TODO: check numbering --> applies. Directories and non-payload files MUST NOT be described by a `file` element.

Reviewer: ```suggestion
   `original-filepaths.txt` exists, rule 2.6.2 applies. Directories and non-payload files MUST NOT be described by a `file` element.
```

3. There MUST NOT be more than one `file` element corresponding to a payload file and every payload file MUST be
   described by a `file` element.

#### 3.3 **(MIGRATION)** migration files in `metadata` folder

1. **(MIGRATION)** The file `metadata/depositor-info/agreements.xml` MUST adhere to [DANS deposit agreement metadata]{:target=_blank}.

2. **(MIGRATION)** The file `metadata/amd.xml` is REQUIRED and MUST adhere to [DANS administrative metadata]{:target=_blank}.

3. **(MIGRATION)** The file `metadata/emd.xml` is optional. If it is present it MUST adhere to
   [DANS EASY metadata schema]{:target=_blank}.

4. **(MIGRATION)** The file `metadata/provenance.xml` is optional. If it is present it MUST adhere
   to [DANS provenance metadata schema]{:target=_blank}.

### 4 Dataset update requirements

> 1. If `bag-info.txt` contains the element `Is-Version-Of` the UUID in its value MUST be a Sword Token 

Reviewer: ```suggestion
1. If `bag-info.txt` contains the element `Is-Version-Of` the UUID in its value MUST be a sword token 
```

9. All URLs used in `metadata/dataset.xml` MUST be valid URLs with protocol http or https.

10. The file `metadata/dataset.xml` MUST have a rights holder, either **(DEPOSIT)** in `<dcterms:rightsHolder>` or **(MIGRATION)** as an author with
    role `RightsHolder`.

#### 3.2 `metadata/files.xml`

1. The file `metadata/files.xml` MUST adhere to [DANS bag file metadata schema]{:target=_blank}.

2. Each `file` element's `filepath` attribute MUST contain the bag local path to the payload file described. When an 
   `original-filepaths.txt` exists, rule 2.7.2<!-- TODO: check numbering --> applies. Directories and non-payload files MUST NOT be described by a `file` element.

3. There MUST NOT be more than one `file` element corresponding to a payload file and every payload file MUST be
   described by a `file` element.

> #### 3.3 **(MIGRATION)** `metadata/depositor-info/agreements.xml`

Author: ```suggestion
#### 3.3 **(MIGRATION)** migration files in `metadata` folder
```
